### ISSUE TITLE
offline render?

### ISSUE THREAD

**Issue by nogweii** [cidrblock/dld4e#6]
I realize that it's mostly front-end, but I have an idea about a tool to automatically generate the appropriate YAML, and then render it. It's not necessarily a problem to have it save the YAML and throw it up onto a webserver, but figured a way to generate the final diagram ahead of time would make things simpler.

**Comment by csillab** [comment]
I was also looking for something like this, rendering them offline than use them embedded in other documentation.

**Comment by tomaskaminskas** [comment]
Another issue is regarding privacy, I would like to generate network diagram with all sensitive information in my private webserver, not in public one.

**Comment by lonniev** [comment]
Perhaps related to this Issue #6: a REST interface to go.drawthe.net would be very handy for automation of technical documentation. One could save the yaml and curl POST it go.drawthe.net to receive back a PNG of the appropriate network diagram.

**Comment by itdependsnetworks** [comment]
What is the privacy issue? Why can't you host this internally? 

There is now a PNG export, and you can simply change the source from a yaml file to a rest call to a bespoke api, that returns the json object, and convert it back to yaml.

In short, I don't think that there is any asks here, that are not supported, but I could be missing the point.

**Comment by lonniev** [comment]
I’m not aware of any issue with privacy.

I’m only curious for a way to call the go.drawthe.net service from a
scripted build.

If the service is just some node/java app that can easily be run on my
local systems then I can bring in it local. I just like the idea of not
having to set up that system as part of my build just to push over a yaml
and get back a png.



-- 
Lonnie VanZandt
303-900-3048

On 14 April 2018 at 15:17:39, Ken Celenza (notifications@github.com) wrote:

What is the privacy issue? Why can't you host this internally?

There is now a PNG export, and you can simply change the source from a yaml
file to a rest call to a bespoke api, that returns the json object, and
convert it back to yaml.

In short, I don't think that there is any asks here, that are not
supported, but I could be missing the point.

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub
<https://github.com/cidrblock/drawthe.net/issues/6#issuecomment-381361092>,
or mute the thread

**Comment by nogweii** [comment]
Indeed, my initial request was just a way to automate the conversion of the YAML to a PNG. Right now, if I wanted to add a network diagram I have to open the YAML in the browser, then save off the PNG. I'm not aware of an automated process.

For those concerned about the privacy aspects, drawthe.net is OSS and could be deployed on their own servers. That is independent of adding an automated conversion.

**Comment by bonndan** [comment]
Any plans to provide a "binary" like a docker image?

**Comment by bonndan** [comment]
Wrote this based on your code:
https://github.com/bonndan/dld4e-headless

Still not complete but one can see where it's going.

**Comment by edurguti** [comment]
> Wrote this based on your code:
> https://github.com/bonndan/dld4e-headless
> 
> Still not complete but one can see where it's going.

Thanks for making this into a server side call, I still can't get notes text to appear, it's just blank.

**Comment by edurguti** [comment]
> What is the privacy issue? Why can't you host this internally?
> 
> There is now a PNG export, and you can simply change the source from a yaml file to a rest call to a bespoke api, that returns the json object, and convert it back to yaml.
> 
> In short, I don't think that there is any asks here, that are not supported, but I could be missing the point.

how can we make an api request to this?

**Comment by sky-joker** [comment]
Hi guys.

I, too, that I wanted to render a network diagram from locally constructed drawthe automatically.  
So I made the automatically render tool of a network diagram from drawthe.

https://github.com/sky-joker/wrapdraw

Maybe it will help you.
